### ISSUE TITLE
Sketcher: Can't join two arcs

### ISSUE THREAD

**Issue by dancesWithMachines** [FreeCAD/FreeCAD#13645]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Scenario:
1. Create one half of the arc and constrain it.
2. Add second arc using same point as center and join the "arms".

Expected results:
Arcs are joined together, sketch is constrained.

Actual result:
Cannot join the arcs.

Visual:
![ScreenRecording2024-04-25at18 16 33-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/0259d030-6b31-4561-8446-c3045fe80791)

### Full version info

```shell
[code]
OS: macOS 14.4
Word size of FreeCAD: 64-bit
Version: 0.21.2.33771 (Git)
Build type: Release
Branch: (HEAD detached at 0.21.2)
Hash: b9bfa5c5507506e4515816414cd27f4851d00489
Python 3.10.13, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.6, OCC 7.6.3
Locale: C/Default (C)
[/code]
```

### Subproject(s) affected?

Sketcher

### Anything else?

I believe this is a regression.
Two arcs cannot be joined in FreeCad 21.0+.
I verified this on 20.2 and it would be over-constrained but work.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
can you check if you have enabled the setting auto remove constraints and play with it on/off?
Could you please also test this in the current dev version: https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds

**Comment by dancesWithMachines** [comment]
> can you check if you have enabled the setting auto remove constraints and play with it on/off?

I had `auto remove redundants` enabled by default. I disabled it and now it behaves like this.
On 21.2 I get over-constrain error like below.
![ScreenRecording2024-04-25at19 01 42-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/41704727-1dc0-43b8-a231-c85f6cbecd06)

On 20.02 I initially get same error, but if I try to move the sketch it resolves.
![ScreenRecording2024-04-25at18 58 49-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/38d1c335-7ce6-4498-a226-d517dbf42784)

> Could you please also test this in the current dev version: https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds

On [newest dev-build](https://github.com/FreeCAD/FreeCAD-Bundle/releases/download/weekly-builds/FreeCAD_weekly-builds-36958-2024-04-23-conda-macOS-x86_64-py311.dmg) (36958-2024-04-23 )  it just crashes and exits.
![ScreenRecording2024-04-25at19 17 45-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/85004c85-b545-49c8-8947-29e985b91e92)

Crash log from devbuild: [freecad_crash.zip](https://github.com/FreeCAD/FreeCAD/files/15113884/freecad_crash.zip)

**Comment by maxwxyz** [comment]
@PaddleStroke FYI

**Comment by dancesWithMachines** [comment]
In this particular scenario `equal` can be used to achieve same result, but this feels like workaround'ing. From logical point of view I don't see why joining two arcs would not work, it isn't intuitional. 
![ScreenRecording2024-04-25at22 56 00-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/d6f6286e-31e6-4d90-a7e9-01ab438ad8bc)
![ScreenRecording2024-04-25at22 56 43-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/45e050c0-0016-40a8-980d-9af66715a676)

**Comment by dancesWithMachines** [comment]
The scenario from this bug is very simple, but I originally come up from this.
![ScreenRecording2024-04-25at23 20 33-ezgif com-video-to-gif-converter](https://github.com/FreeCAD/FreeCAD/assets/44846363/c2b99fd5-9304-4d8c-a890-a7de73e9750d)
Here the outcome is different, depending on what I do first.

**Comment by PaddleStroke** [comment]
I totally agree that this feels very much like a bug. I have argued that in the past.
I'm just not sure how to solve it. Anyone know how other Cads are handling this situation? 

Maybe we need to add an exceptions in the solver for coincidence constraints for cases where arcs have 3 coincidences. Something like : 
```
if (arc has 3 coincidence)
      ignore one dof of center coincidence
```

**Comment by Shkolik** [comment]
> Maybe we need to add an exceptions in the solver for coincidence constraints for cases where arcs have 3 coincidences. 

Sound like a good starting point. It's an edge case, but very frequently used one..

**Comment by luzpaz** [comment]
* Related to #5848 ?

**Comment by PaddleStroke** [comment]
@FlachyJoe any idea? What do you think of the suggestion I made above : 

> Maybe we need to add an exceptions in the solver for coincidence constraints for cases where arcs have 3 coincidences. Something like :
> 
> if (arc has 3 coincidence)
>       ignore one dof of center coincidence

@maxwxyz do you know how this behaves in other CADs?

**Comment by FlachyJoe** [comment]
@PaddleStroke IMHO we shouldn't do such exception in solver.
What can be done alternatively is to replace a coincidence by an horizontal or vertical constraint at arc or constraint creation i.e. in Command or GUI code.

**Comment by PaddleStroke** [comment]
@pierreporte could you please feedback how this is handled by SW/catia/fusion... ?

**Comment by pierreporte** [comment]
@PaddleStroke In Catia the sur ch remains constrained as expected. There is no reason FreeCAD should behave as it does.
